Makefile

```diff
@@ -3,7 +3,7 @@ SDIR=src
 TDIR=pytests
 # compilers
 TSC=./node_modules/typescript/bin/tsc
-TSCFLAGS=-t ES3 --sourceMap --module commonjs --noImplicitAny
```

Comment:
I was having a problem determining the type of the ```err``` argument in the ```domain.on(..., err)```. I have found a workaround to this, in my latest commit.

Also, on a clean project (after doing ```make clean```), I have to run ```make test``` three times before the makefile stops terminating with an error, and begins execution of tests. I still haven't found a work around to this.